build-system/app-index/index.js

```diff
@@ -59,97 +49,78 @@ function renderListing(basepath) {
     return Promise.resolve(null);
   }
 
-  return fs.statAsync(path).then(stat => {
-    if (!stat.isDirectory()) {
-      return null;
-    }
-
-    return Promise.all([
-      fs.readdirAsync(path),
-      fs.readFileAsync(templateFile),
-      fs.readFileAsync(mainCssFile),
-    ]).then(result => {
-      const files = result[0];
-      const template = result[1].toString();
-      const css = result[2].toString();
-
-      return template
-          .replace('<!-- main_style -->', `<style>${css}</style>`)
-          .replace('<!-- basepath -->', basepath)
-          .replace('<!-- listing -->',
-              files.map(file => renderFileLink(basepath, file)).join(''));
-    });
-  }).catch(() => /* empty catch for fallbacks */ null);
+  return fs.statAsync(path).then(stat =>
```

Comment:
A little `async/await` would make this clearer.

Reply:
👍